Source/Charts/Data/Implementations/Standard/ChartDataSet.swift

```diff
@@ -86,22 +86,22 @@ open class ChartDataSet: ChartBaseDataSet
     
     open override func calcMinMax()
     {
-        guard !values.isEmpty else { return }
-
         _yMax = -Double.greatestFiniteMagnitude
         _yMin = Double.greatestFiniteMagnitude
         _xMax = -Double.greatestFiniteMagnitude
         _xMin = Double.greatestFiniteMagnitude
 
+        guard !values.isEmpty else { return }
+
         values.forEach { calcMinMax(entry: $0) }
     }
     
     open override func calcMinMaxY(fromX: Double, toX: Double)
     {
-        guard !values.isEmpty else { return }
-
         _yMax = -Double.greatestFiniteMagnitude
         _yMin = Double.greatestFiniteMagnitude
+
+        guard !values.isEmpty else { return }
```

Comment:
The change here seems legit. But how you find `calcMinMaxY` needs fix as well? I don't see `clear()` will eventually call `calcMinMaxY`?
I just want to make sure we don't missed other similar issues. Can you help do a double check?

Reply:
You're right, `calcMinMaxY` isn't triggered when calling `clear()`, and I haven't had any trouble from its current implementation, but I thought it would still be better to reset `_yMax` and `_yMin` to their defaults rather than leaving them with random numbers from the previous set. Feel free to edit my changes if you'd rather leave that alone.

Reply:
alright, got you. seems fine to me.

Reply:
It's weird that this issue is only coming up now, as the implementation hasn't changed in a very long time. @carlo I think you're right and that the guard statement should be removed entirely.